parser/src/aspects/type.rs

```diff
@@ -78,11 +78,7 @@ impl<'a> TypeAspect<'a> for Parser<'a> {
     }
 
     fn parse_type_parameter_list(&mut self) -> ParseResult<(Vec<Type<'a>>, SourceSegment)> {
-        if self
-            .cursor
-            .lookahead(blanks().then(of_type(SquaredLeftBracket)))
-            .is_none()
-        {
+        if self.cursor.lookahead(of_type(SquaredLeftBracket)).is_none() {
```

Comment:
Since this parser is also used for function declarations, it should be noted that `fun foo [T]() = {}` is now an invalid input if not updated.

Reply:
I think that the fact main parser methods will automatically remove blanks / spaces at the begining is a conception issue that should be removed in a later PR, I fixed it here 'cause i needed more control over heading blanks in `any_call` but indeed i haven't anticipated the  change needed to by other methods to remove heading blanks before calling it.